src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -426,19 +429,37 @@ private static void makePublic(final CtClass c) {
         boolean foundDefault = false;
         if (!cl.isInterface()) {
           for (CtConstructor ctor : cl.getConstructors()) {
-            if (ctor.getParameterTypes().length == 0
-                && (ctor.getModifiers() & javassist.Modifier.ABSTRACT) == 0
-                && !ctor.isEmpty()) {
+            if (ctor.isEmpty() && ctor.getParameterTypes().length == 0
+                && (ctor.getModifiers() & javassist.Modifier.ABSTRACT) == 0) {
               foundDefault = true;
               break;
             }
           }
         }
-        if (!foundDefault) {
+        if (cl.isInterface()) {
           CtConstructor newCtor =
               new CtConstructor(new CtClass[] {}, implementation);
           newCtor.setBody("{}");
           implementation.addConstructor(newCtor);
+        } else if (!foundDefault) {
```

Comment:
Given how complicated this code has got, I suggest writing some unit tests for this method.